public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -192,10 +195,53 @@ export function SpanDetailTable(props: SpanDetailTableProps) {
     ).finally(() => setIsSpansTableDataLoading(false));
   };
 
+  // For application_analytics
   useEffect(() => {
-    fetchData();
+    if (!props.payloadData) {
+      fetchData();
+    }
   }, [tableParams, props.DSL]);
 
+  useEffect(() => {
+    if (!props.payloadData) {
+      console.warn('No payloadData provided in SpanDetailTable');
+      return;
+    }
+    try {
+      const parsed = JSON.parse(props.payloadData);
+
+      const hitsArray =
+        parsed.hits && Array.isArray(parsed.hits.hits)
+          ? parsed.hits.hits
+          : Array.isArray(parsed)
+          ? parsed
+          : [];
+
+      if (hitsArray.length === 0) {
+        console.warn('No hits found in payloadData');
```

Comment:
Why are we doing this, the response should be the same as the request doesn't change.
```
const hitsArray =
        parsed.hits && Array.isArray(parsed.hits.hits)
          ? parsed.hits.hits
          : Array.isArray(parsed)
          ? parsed
          : [];
 ```
Also, Do we need `console.warn`?

Reply:
Removed console.warn 
I think the fallback to prevent page crashing is helpful although the request will not be changing. Just an extra safe-guard